test/python/test_select.py

```diff
@@ -64,3 +64,20 @@ def run_test(sess):
 
         assert (
             self.with_ngraph(run_test) == self.without_ngraph(run_test)).all()
+
+    def test_select_complexshape(self):
+        a = [1, 0, 1, 0, 1, 1, 0]
+        x = [[[[4], [2], [9]], [[3], [8], [2]]],
+             [[[6], [6], [2]], [[5], [4], [1]]],
+             [[[7], [6], [0]], [[1], [1], [1]]],
+             [[[5], [0], [9]], [[8], [7], [6]]],
+             [[[1], [9], [6]], [[3], [2], [8]]],
+             [[[9], [2], [5]], [[3], [3], [6]]],
+             [[[4], [9], [3]], [[6], [1], [4]]]]
```

Comment:
please add a `# comment` beside each line indicating their shapes
Also the predicate in this test case is of shape [7]

we need more combinations (eg, [7 2 1] as predicate shape, [5 3 7 2 1] as x, y shape

Also you dont need to hand populate x, just use np's random function or np.ones([3,4,5]) etc